### ISSUE TITLE
could not deploy local blog to github 

### ISSUE THREAD

**Issue by vulgatecnn** [hexojs/hexo#2167]
> before you submit your issue, please delete all the example code in template
## Environment Info

Node version(`node -v`)

```
v4.5.0
```

Your site `_config.yml`
# Hexo Configuration
## Source: https://github.com/hexojs/hexo/
# Site

title: Hexo
subtitle:
description:
author: John Doe
language:
timezone:
# URL
## If your site is put in a subdirectory, set url as 'http://yoursite.com/child' and root as '/child/'

url: http://yoursite.com
root: /
permalink: :year/:month/:day/:title/
permalink_defaults:
# Directory

source_dir: source
public_dir: public
tag_dir: tags
archive_dir: archives
category_dir: categories
code_dir: downloads/code
i18n_dir: :lang
skip_render:
# Writing

new_post_name: :title.md # File name of new posts
default_layout: post
titlecase: false # Transform title into titlecase
external_link: true # Open external links in new tab
filename_case: 0
render_drafts: false
post_asset_folder: false
relative_link: false
future: true
highlight:
  enable: true
  line_number: true
  auto_detect: false
  tab_replace:
# Category & Tag

default_category: uncategorized
category_map:
tag_map:
# Date / Time format
## http://momentjs.com/docs/#/displaying/format/

date_format: YYYY-MM-DD
time_format: HH:mm:ss
# Pagination
## Set per_page to 0 to disable pagination

per_page: 10
pagination_dir: page
# Extensions
## Themes: https://hexo.io/themes/

theme: landscape
# Deployment
## Docs: https://hexo.io/docs/deployment.html

deploy:
  type: git
  respository: https://github.com/vulgatecnn/blog.git
  branch: master
  message: update

'''

Plugin version(`npm ls --depth 0`)

```
hexo-site@0.0.0 /Users/xxx/code/hexo/blog
├── hexo@3.2.2
├── hexo-deployer-git@0.2.0
├── hexo-generator-archive@0.1.4
├── hexo-generator-category@0.1.3
├── hexo-generator-index@0.2.0
├── hexo-generator-tag@0.2.0
├── hexo-renderer-ejs@0.2.0
├── hexo-renderer-marked@0.2.11
├── hexo-renderer-stylus@0.3.1
└── hexo-server@0.2.0
```
## For BUG
- BUG description
- the way to reproduce
- log with hexo d --debug

02:32:13.595 DEBUG Hexo version: 3.2.2
02:32:13.598 DEBUG Working directory: ~/code/hexo/blog/
02:32:13.675 DEBUG Config loaded: ~/code/hexo/blog/_config.yml
02:32:13.685 DEBUG Script loaded: themes/landscape/scripts/fancybox.js
02:32:13.896 DEBUG Plugin loaded: hexo-deployer-git
02:32:13.899 DEBUG Plugin loaded: hexo-generator-archive
02:32:13.901 DEBUG Plugin loaded: hexo-generator-category
02:32:13.904 DEBUG Plugin loaded: hexo-generator-index
02:32:13.906 DEBUG Plugin loaded: hexo-generator-tag
02:32:13.909 DEBUG Plugin loaded: hexo-renderer-ejs
02:32:13.987 DEBUG Plugin loaded: hexo-renderer-marked
02:32:13.989 DEBUG Plugin loaded: hexo-renderer-stylus
02:32:14.087 DEBUG Plugin loaded: hexo-server
02:32:14.090 INFO  Deploying: git
You have to configure the deployment settings in _config.yml first!

Example:
  deploy:
    type: git
    repo: <repository url>
    branch: [branch]
    message: [message]

For more help, you can check the docs: http://hexo.io/docs/deployment.html
02:32:14.092 INFO  Deploy done: git
02:32:14.095 DEBUG Database saved
## For question

could not deploy local blog to github

**Comment by vulgatecnn** [comment]
_config.yml

# Hexo Configuration
## Docs: https://hexo.io/docs/configuration.html
## Source: https://github.com/hexojs/hexo/

# Site
title: Hexo
subtitle:
description:
author: John Doe
language:
timezone:

# URL
## If your site is put in a subdirectory, set url as 'http://yoursite.com/child' and root as '/child/'
url: http://yoursite.com
root: /
permalink: :year/:month/:day/:title/
permalink_defaults:

# Directory
source_dir: source
public_dir: public
tag_dir: tags
archive_dir: archives
category_dir: categories
code_dir: downloads/code
i18n_dir: :lang
skip_render:

# Writing
new_post_name: :title.md # File name of new posts
default_layout: post
titlecase: false # Transform title into titlecase
external_link: true # Open external links in new tab
filename_case: 0
render_drafts: false
post_asset_folder: false
relative_link: false
future: true
highlight:
  enable: true
  line_number: true
  auto_detect: false
  tab_replace:

# Category & Tag
default_category: uncategorized
category_map:
tag_map:

# Date / Time format
## Hexo uses Moment.js to parse and display date
## You can customize the date format as defined in
## http://momentjs.com/docs/#/displaying/format/
date_format: YYYY-MM-DD
time_format: HH:mm:ss

# Pagination
## Set per_page to 0 to disable pagination
per_page: 10
pagination_dir: page

# Extensions
## Plugins: https://hexo.io/plugins/
## Themes: https://hexo.io/themes/
theme: landscape

# Deployment
## Docs: https://hexo.io/docs/deployment.html
deploy:
  type: git
  respository: https://github.com/vulgatecnn/blog.git
  branch: master
  message: update

**Comment by vulgatecnn** [comment]
# Hexo Configuration
## Docs: https://hexo.io/docs/configuration.html
## Source: https://github.com/hexojs/hexo/

# Site
title: Hexo
subtitle:
description:
author: John Doe
language:
timezone:

# URL
## If your site is put in a subdirectory, set url as 'http://yoursite.com/child' and root as '/child/'
url: http://yoursite.com
root: /
permalink: :year/:month/:day/:title/
permalink_defaults:

# Directory
source_dir: source
public_dir: public
tag_dir: tags
archive_dir: archives
category_dir: categories
code_dir: downloads/code
i18n_dir: :lang
skip_render:

# Writing
new_post_name: :title.md # File name of new posts
default_layout: post
titlecase: false # Transform title into titlecase
external_link: true # Open external links in new tab
filename_case: 0
render_drafts: false
post_asset_folder: false
relative_link: false
future: true
highlight:
  enable: true
  line_number: true
  auto_detect: false
  tab_replace:

# Category & Tag
default_category: uncategorized
category_map:
tag_map:

# Date / Time format
## Hexo uses Moment.js to parse and display date
## You can customize the date format as defined in
## http://momentjs.com/docs/#/displaying/format/
date_format: YYYY-MM-DD
time_format: HH:mm:ss

# Pagination
## Set per_page to 0 to disable pagination
per_page: 10
pagination_dir: page

# Extensions
## Plugins: https://hexo.io/plugins/
## Themes: https://hexo.io/themes/
theme: landscape

# Deployment
## Docs: https://hexo.io/docs/deployment.html
deploy:
  type: git
  respository: https://github.com/xxx/blog.git
  branch: master
  message: update

**Comment by NoahDragon** [comment]
> before you submit your issue, please delete all the example code in template

Could you rephrase your issue? I couldn't find useful information with samples mixed.
